aesara/tensor/elemwise_cgen.py

```diff
@@ -316,20 +357,32 @@ def make_reordered_loop(
     # more are not broadcasted, in which case the number of elements
     # of any of them will be equal to the number of iterations we have
     # to do.
-    totals = []
+    declare_totals = f"int init_totals[{nnested}];\n"
     for i, candidates in enumerate(zip(*init_loop_orders)):
-        for j, candidate in enumerate(candidates):
-            if candidate != "x":
-                var = sub[f"lv{int(j)}"]
-                total = f"{var}_n{candidate}"
-                break
-        else:
-            total = "1"
-        totals.append(total)
+        nonx_canditates = tuple(
+            (idx, c) for idx, c in enumerate(candidates) if c != "x"
+        )
 
-    declare_totals = f"""
-    int init_totals[{nnested}] = {{{", ".join(totals)}}};
-    """
+        # All inputs are known to be broadcastable
+        if not nonx_canditates:
+            declare_totals += f"init_totals[{i}] = 1;\n"
+            continue
+
+        # There is only one informative source of size
+        if len(nonx_canditates) == 1:
+            idx, candidate = nonx_canditates[0]
+            var = sub[f"lv{int(idx)}"]
+            declare_totals += f"init_totals[{i}] = {var}_n{candidate};\n"
+            continue
+
+        # In this case any non-size 1 variable will define the right size
+        declare_totals += f"init_totals[{i}] = "
+        for (idx, candidate) in nonx_canditates[:-1]:
+            var = sub[f"lv{int(idx)}"]
+            declare_totals += f"({var}_n{candidate} != 1)? {var}_n{candidate}: "
+        idx, candidate = nonx_canditates[-1]
+        var = sub[f"lv{idx}"]
+        declare_totals += f"{var}_n{candidate};\n"
```

Comment:
This block looks very similar to the block of code added above, which implies that it should probably be abstracted to a function.

Reply:
I would keep it dumb for now. I have strong prior that we will have to touch this code again very soon

Reply:
> I would keep it dumb for now. I have strong prior that we will have to touch this code again very soon

Reducing duplication helps even more when code needs to undergo change (e.g. one only needs change something once, instead of twice).

Reply:
Done